pymc3_hmm/model_utils.py

```diff
@@ -0,0 +1,36 @@
+from pymc3_hmm.distributions import HMMStateSeq
+from pymc3_hmm.utils import compute_steady_state
+import theano.tensor as tt
+import pymc3 as pm
+import numpy as np
+
+
+def auto_set_hmm_seq(N_states, model, states):
+    """
+    Initiate a HMMStateSeq based on the length of the mixture component.
+
+    This function require pymc3 and HMMStateSeq.
+
+    Parameters
+    ----------
+    N_states : int
+        Number of states in the mixture
+    model : pymc3.model.Model
+        Model object that we trained on
+    states : ndarray
+        Vector sequence of states to set the `test_value` for `HMMStateSeq`
+
+    Returns
+    -------
+    locals(), a dict of local variables for reference in sampling steps.
+    """
+    with model:
```

Comment:
We can make the `model` parameter optional (with a default of `None`) if we use `model = pm.modelcontext(model)` before this line.  `pm.modelcontext` will get the model from the surrounding `with`-context, if any, or use the given `model` when it's non-`None`.